docs/dev/guidelines/criteria-builder.rst

```diff
@@ -0,0 +1,294 @@
+
+    .. code-block:: java
+
+        @Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
+        @StaticMetamodel(User.class)
+        public abstract class User_ extends de.tum.in.www1.artemis.domain.AbstractAuditingEntity_ {
+
+            public static volatile SingularAttribute<User, String> lastName;
+            public static volatile SingularAttribute<User, Instant> resetDate;
+            public static volatile SingularAttribute<User, ZonedDateTime> hideNotificationsUntil;
+            public static volatile SetAttribute<User, String> groups;
+            public static volatile SetAttribute<User, GuidedTourSetting> guidedTourSettings;
+            public static volatile SingularAttribute<User, String> login;
+            public static volatile SingularAttribute<User, String> activationKey;
+            public static volatile SingularAttribute<User, String> resetKey;
+            public static volatile SetAttribute<User, Authority> authorities;
+            ...
+        }
+
+2. **JpaSpecificationExecutor:** To execute Specifications and generate SQL statements, we need to extend the JpaSpecificationExecutor interface in our Spring Data JPA Repository.
+
+    .. code-block:: java
+
+        @Repository
+        public interface UserRepository extends JpaRepository<User, Long>, JpaSpecificationExecutor<User> {
+        }
+
+3. **(Optional) Show queries:** To ease debugging, generated queries can be displayed by enabling the output of executed SQL probes.
+
+    .. code-block:: yaml
+
+        jpa:
+            database-platform: org.hibernate.dialect.MySQL8Dialect
+            database: MYSQL
+            show-sql: true
+
+
+2. Generating the query
+=======================
+
+1. **Query Generation:** In most occasions, it is sufficient to execute any one of the following methods:
+
+    .. code-block:: java
+
+        List<T> findAll(Specification<T> spec);
+
+        Page<T> findAll(Specification<T> spec, Pageable pageable);
+
+        List<T> findAll(Specification<T> spec, Sort sort);
+
+2. **Defining the initial Specification:** To generate a query with multiple Specifications, we can use the `and()` method for concatenation. However, the first Specification must always be called via the `where()` method as a rule.
+
+    .. code-block:: java
+
+        Specification<T> specification = Specification.where(getFirstSpecification()).and(getSecondSpecification()).and(getThirdSpecification())...and(getNthSpecification());
+        return findAll(specification, sort/pageable);
+
+3. **Defining Specifications:** A specification is a functional interface with a single method. This method has three parameter - a root, a query and a criteria builder. You don't need to specify these arguments manually because they are provided during chaining.
```

Comment:
```suggestion
3. **Defining Specifications:** A specification is a functional interface with a single method. This method has three parameters - a root, a query and a criteria builder. You don't need to specify these arguments manually because they are provided during chaining.
```